dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.h

```diff
@@ -64,6 +65,47 @@ typedef RcHandle<RtpsUdpInst> RtpsUdpInst_rch;
 typedef RcHandle<RtpsUdpTransport> RtpsUdpTransport_rch;
 typedef RcHandle<TransportClient> TransportClient_rch;
 
+struct LocatorCacheKey {
+  LocatorCacheKey(const RepoId& remote, const RepoId& local, bool prefer_unicast) : remote_(remote), local_(local), prefer_unicast_(prefer_unicast) {}
+  bool operator<(const LocatorCacheKey& rhs) const {
+    return std::memcmp(this, &rhs, sizeof (LocatorCacheKey)) < 0;
+  }
+
+  const RepoId remote_;
+  const RepoId local_;
+  const bool prefer_unicast_;
+};
+typedef AddressCache<LocatorCacheKey> LocatorCache;
+
+struct BundlingCacheKey {
+  BundlingCacheKey(const RepoId& dst_guid, const RepoId& from_guid, const RepoIdSet& to_guids) : dst_guid_(dst_guid), from_guid_(from_guid), to_guids_(to_guids) {}
+  bool operator<(const BundlingCacheKey& rhs) const {
+    int r1 = std::memcmp(&dst_guid_, &rhs.dst_guid_, sizeof (RepoId));
```

Comment:
For this one in particular, I was actually thinking I'd combine the memcmp for the first two guids, since the behavior should be the same. But your comment about packing below reminds me I need to make that explicit.